Starbucks/Controllers/ProductsViewController/ProductsViewController.swift

```diff
@@ -8,10 +8,78 @@
+class ProductsViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
     
+    let spacing: CGFloat = 2
+    let products: [Product]
+    
+    lazy var productCollectionView: UICollectionView = {
+        let flowLayout = UICollectionViewFlowLayout()
+        flowLayout.scrollDirection = .vertical
+        flowLayout.minimumLineSpacing = spacing
+        flowLayout.minimumInteritemSpacing = spacing
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: flowLayout)
+        collectionView.isPagingEnabled = true
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+    
+    init(products: [Product]) {
+        self.products = products
+        super.init(nibName: nil, bundle: nil)
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = .green
+        view.backgroundColor = UIColor.white
+        setupConstraints()
+        registerCollectionCells()
+    }
+    
+    func setupConstraints() {
+        view.addSubview(productCollectionView)
+        
+        productCollectionView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 20).isActive = true
+        productCollectionView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -20).isActive = true
+        productCollectionView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        productCollectionView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
     }
+    
+    func registerCollectionCells() {
+        productCollectionView.register(ProductCollectionViewCell.self, forCellWithReuseIdentifier: "ProductCollectionViewCellID")
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return products.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ProductCollectionViewCellID", for: indexPath) as! ProductCollectionViewCell
+        let product = products[indexPath.item]
+        cell.productImageView.image = product.image
+        cell.productNameLabel.text = product.name
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let numberOfItemsPerRow: CGFloat = 2
+        let spacingBetweenItems: CGFloat = 8
+        let totalSpacing = (2 * spacingBetweenItems) + ((numberOfItemsPerRow - 1) * spacingBetweenItems)
+        let width = (collectionView.bounds.width - totalSpacing) / numberOfItemsPerRow
+        return CGSize(width: 171, height: 230)
```

Comment:
@RSKuse I noticed you calculated the width and height but did not use it. You hard coded the CGSize.

Reply:
The result of the hard coding when I run the build on a iPhoneSE. @RSKuse 

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-05-06 at 17 35 44](https://github.com/RSKuse/Starbucks/assets/5164871/558242af-2c92-43f2-bfa2-c5e91ad1f3c4)

Reply:
I do not understand here? what you  mean